editor/src/components/inspector/sections/layout-section/layout-system-subsection/split-chained-number-input.tsx

```diff
@@ -146,44 +152,160 @@ function getInitialMode(
   return defaultMode
 }
 
-function onSubmitValue(old: ControlCSSNumber) {
-  return wrappedEmptyOrUnknownOnSubmitValue(old.onSubmitValue, old.onUnsetValues)
-}
+export type FourValue =
+  | { type: 'T'; value: CSSNumber }
+  | { type: 'R'; value: CSSNumber }
+  | { type: 'B'; value: CSSNumber }
+  | { type: 'L'; value: CSSNumber }
+
+export type TwoValue = { type: 'V'; value: CSSNumber } | { type: 'H'; value: CSSNumber }
 
-function onTransientSubmitValue(old: ControlCSSNumber) {
-  return wrappedEmptyOrUnknownOnSubmitValue(old.onTransientSubmitValue, old.onUnsetValues)
+export type SplitChainedEvent =
+  | { type: 'one-value'; value: CSSNumber }
+  | { type: 'two-value'; value: TwoValue }
+  | { type: 'four-value'; value: FourValue }
+
+export type SplitControlValues = {
+  oneValue: CSSNumber | null
+  horizontal: CSSNumber | null
+  vertical: CSSNumber | null
+  top: CSSNumber
+  right: CSSNumber
+  bottom: CSSNumber
+  left: CSSNumber
 }
 
-const onSubmitValueShorthand =
+const handleSplitChainedEvent =
   (
-    update: (v: CSSNumber) => void,
-    updateShorthand: UpdateShorthand | null,
-    sides: Array<ControlCSSNumber>,
-    keepSides: Partial<Sides>,
-    transient?: boolean,
+    e: SplitChainedEvent,
+    dispatch: EditorDispatch,
+    element: ElementPath,
+    shorthand: PropertyPath,
+    longhand: {
+      T: PropertyPath
+      R: PropertyPath
+      B: PropertyPath
+      L: PropertyPath
+    },
   ) =>
-  (input: UnknownOrEmptyInput<CSSNumber>) => {
-    if (!isCSSNumber(input)) {
-      return
-    }
-    update(input)
-    if (updateShorthand != null) {
-      updateShorthand(
-        {
-          top: keepSides.top ?? input,
-          left: keepSides.left ?? input,
-          bottom: keepSides.bottom ?? input,
-          right: keepSides.right ?? input,
-        },
-        transient,
+  (useShorthand: boolean, aggregates: SplitControlValues): void => {
+    const setProp = (path: PropertyPath, values: (CSSNumber | null)[]): EditorAction => {
+      return setProp_UNSAFE(
+        element,
+        path,
+        jsxAttributeValue(
+          mapDropNulls((v) => v, values)
+            .map((v) => printCSSNumber(v, null))
```

Comment:
this is where you want to replace `null` with `'px'` to omit printing the `'px'`  from simple values such as `'20px'`!

Reply:
oooooh nice! thank you 🙏 🙏